extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultHealthExtension.java

```diff
@@ -9,61 +9,51 @@
 import org.eclipse.edc.runtime.metamodel.annotation.Requires;
-import org.eclipse.edc.runtime.metamodel.annotation.Setting;
-import org.eclipse.edc.spi.http.EdcHttpClient;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 import org.eclipse.edc.spi.system.health.HealthCheckService;
-import org.eclipse.edc.spi.types.TypeManager;
+
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_HEALTH_CHECK_ENABLED;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultConfig.VAULT_HEALTH_CHECK_ENABLED_DEFAULT;
 
 @Requires(HealthCheckService.class)
+@Extension(value = HashicorpVaultHealthExtension.NAME)
 public class HashicorpVaultHealthExtension implements ServiceExtension {
-    public static final boolean VAULT_HEALTH_CHECK_DEFAULT = true;
 
-    @Setting(value = "Whether or not the vault health check is enabled", defaultValue = "true", type = "boolean")
-    public static final String VAULT_HEALTH_CHECK = "edc.vault.hashicorp.health.check.enabled";
+    public static final String NAME = "Hashicorp Vault Health";
 
     @Inject
     private HealthCheckService healthCheckService;
 
     @Inject
-    private TypeManager typeManager;
-
-    @Inject
-    private EdcHttpClient httpClient;
+    private HashicorpVaultClient client;
 
     @Override
     public String name() {
-        return "Hashicorp Vault Health Check";
+        return NAME;
     }
 
-
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var config = HashicorpVaultClientConfig.create(context);
-        var client = new HashicorpVaultClient(config, httpClient, typeManager.getMapper());
-        configureHealthCheck(client, context);
-
-        context.getMonitor().info("HashicorpVaultExtension: health check initialization complete.");
-    }
-
-    private void configureHealthCheck(HashicorpVaultClient client, ServiceExtensionContext context) {
-        var healthCheckEnabled =
-                context.getSetting(VAULT_HEALTH_CHECK, VAULT_HEALTH_CHECK_DEFAULT);
-        if (!healthCheckEnabled) return;
-
-        var healthCheck =
-                new HashicorpVaultHealthCheck(client, context.getMonitor());
-
-        healthCheckService.addLivenessProvider(healthCheck);
-        healthCheckService.addReadinessProvider(healthCheck);
-        healthCheckService.addStartupStatusProvider(healthCheck);
+        var monitor = context.getMonitor().withPrefix(NAME);
+        var healthCheckEnabled = context.getSetting(VAULT_HEALTH_CHECK_ENABLED, VAULT_HEALTH_CHECK_ENABLED_DEFAULT);
+        if (healthCheckEnabled) {
+            var healthCheck = new HashicorpVaultHealthCheck(client, monitor);
+            healthCheckService.addLivenessProvider(healthCheck);
+            healthCheckService.addReadinessProvider(healthCheck);
+            healthCheckService.addStartupStatusProvider(healthCheck);
+            monitor.info("Healthcheck initialization complete");
```

Comment:
Please change this to debug level and be more descriptive, e.g. "Vault health check." Even though the message is prefixed with the extension name. Also, I think health check is two words.